tests/daemon/test-daemon.c

```diff
@@ -1225,48 +1175,13 @@ test_daemon_does_not_record_sequences_when_daemon_disabled (Fixture      *fixtur
 }
 
 static void
-test_daemon_updates_timestamps_on_shutdown (Fixture      *fixture,
```

Comment:
I agree with that general statement, but it's not the memory management that we're interested in testing here; it's the accuracy of the timestamps, which is improved when the persistent cache is properly finalized. The fact that the update timestamps call happens during finalization is purely a coincidence in moral terms.